pydantic/_internal/_validators.py

```diff
@@ -386,10 +387,11 @@ def max_digits_validator(x: Any, max_digits: Any) -> Any:
 
 
 def decimal_places_validator(x: Any, decimal_places: Any) -> Any:
-    decimal_places_, _ = _extract_decimal_digits_info(x)
-    normalized_decimal_places, _ = _extract_decimal_digits_info(x.normalize())
-
     try:
+        if not isinstance(x, Decimal):
+            raise TypeError(f'Expected Decimal, received {type(x).__name__}')
```

Comment:
Do we test this anywhere?

Reply:
I end up checking the Decimal type here instead
https://github.com/pydantic/pydantic/blob/f6919b5ded203cf390d3df7863cbbf1022c3b7cf/pydantic/_internal/_validators.py#L374